From: Kai Liu <kai.liu@suse.com>
Date: Thu, 28 Apr 2022 12:45:32 +0800
Subject: [PATCH] Revert "tracing: Dump stacktrace trigger to the corresponding
 instance"
Patch-mainline: Never, revert stable patch which is added again in sorted section
References: bsn#83
Modified-by-SEL: No

This reverts commit 5e8446e3820c67b644573f7d1387d22dc41ef71e.

Signed-off-by: Kai Liu <kai.liu@suse.com>
---
 kernel/trace/trace_events_trigger.c | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/kernel/trace/trace_events_trigger.c b/kernel/trace/trace_events_trigger.c
index 4bc90965abb2..d0309de2f84f 100644
--- a/kernel/trace/trace_events_trigger.c
+++ b/kernel/trace/trace_events_trigger.c
@@ -1219,14 +1219,7 @@ static void
 stacktrace_trigger(struct event_trigger_data *data, void *rec,
 		   struct ring_buffer_event *event)
 {
-	struct trace_event_file *file = data->private_data;
-	unsigned long flags;
-
-	if (file) {
-		local_save_flags(flags);
-		__trace_stack(file->tr, flags, STACK_SKIP, preempt_count());
-	} else
-		trace_dump_stack(STACK_SKIP);
+	trace_dump_stack(STACK_SKIP);
 }
 
 static void
-- 
2.35.1

